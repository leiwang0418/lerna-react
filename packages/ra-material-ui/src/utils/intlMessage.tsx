const intlMessage: any = (locale: string) => {
	switch (locale) {
		case "en":
			return import("../compiled-lang/zh.json");
		default:
			return import("../compiled-lang/zh.json");
	}
};

export default intlMessage;
