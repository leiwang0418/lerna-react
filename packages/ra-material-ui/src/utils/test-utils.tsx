import React, { FC } from "react";
import { render, RenderOptions } from "@testing-library/react";

// const customRender = (
// 	ui: React.ReactElement,
// 	{...renderOptions } = {}
// ) => {
// 	const Wrapper: FC = ({ children }) => (
// 		<IntlProvider locale={locale} messages={zh}>
// 			<Provider store={store}>
// 				<Router>{children}</Router>
// 			</Provider>
// 		</IntlProvider>
// 	);

// 	return render(ui, { wrapper: Wrapper, ...renderOptions });
// };
const customRender = render;

export * from "@testing-library/react";

export { customRender as render };
