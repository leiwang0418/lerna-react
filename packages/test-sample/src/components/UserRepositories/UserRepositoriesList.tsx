import React, { FC } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import { useIntl, defineMessages } from 'react-intl';

interface Props {
	hasError?: boolean;
}

const messages = defineMessages({
	error: {
		defaultMessage: '加载用户仓库列表失败',
		description: '加载用户仓库列表错误',
	},
	test: {
		defaultMessage: '加载用户仓库列表失败',
		description: '加载用户仓库列表错误',
	},
});

const UserRepositoriesList: FC<Props> = ({ hasError }) => {
	const intl = useIntl();

	if (hasError) {
		return (
			<Typography variant="body1" color="error" gutterBottom>
				{intl.formatMessage(messages.error)}
			</Typography>
		);
	}

	return (
		<List>
			-----
			<Typography variant="body1" color="error" gutterBottom>
				{intl.formatMessage(messages.test)}
			</Typography>
		</List>
	);
};

UserRepositoriesList.propTypes = {
	hasError: PropTypes.bool,
};

export default UserRepositoriesList;
