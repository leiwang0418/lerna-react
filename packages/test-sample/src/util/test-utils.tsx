import React, { FC } from 'react';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import zh from '../compiled-lang/zh.json';

const render = (
	ui: React.ReactElement,
	{ locale = 'zh', ...renderOptions } = {}
) => {
	const Wrapper: FC = ({ children }) => (
		<IntlProvider locale={locale} messages={zh}>
			{children}
		</IntlProvider>
	);

	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';

export { render };
