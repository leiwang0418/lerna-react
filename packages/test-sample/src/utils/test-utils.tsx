import React, { FC } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import zh from "../compiled-lang/zh.json";
import { Provider } from "react-redux";
import { rootReducer } from "../store";

import { BrowserRouter as Router } from "react-router-dom";
import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";

export const configStore = () => {
	return createStore(combineReducers(rootReducer), applyMiddleware(thunk));
};

const customRender = (
	ui: React.ReactElement,
	{ locale = "zh", store = configStore(), ...renderOptions } = {}
) => {
	const Wrapper: FC = ({ children }) => (
		<IntlProvider locale={locale} messages={zh}>
			<Provider store={store}>
				<Router>{children}</Router>
			</Provider>
		</IntlProvider>
	);

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from "@testing-library/react";

export { customRender as render };
export const store = configStore();
