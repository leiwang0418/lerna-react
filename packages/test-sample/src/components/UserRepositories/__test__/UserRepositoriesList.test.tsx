import React from 'react';
import { render } from '../../../utils/test-utils';
import UserRepositoriesList from '../UserRepositoriesList';

describe('snaphost test', () => {
	it('renders the basic UserRepositoriesList', () => {
		const { container } = render(<UserRepositoriesList />);
		
		expect(container).toMatchSnapshot();
	});
});
