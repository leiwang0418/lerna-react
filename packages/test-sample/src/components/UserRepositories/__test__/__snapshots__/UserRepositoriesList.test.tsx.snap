// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snaphost test renders the basic UserRepositoriesList 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          未发现仓库数据
        </span>
      </div>
    </li>
  </ul>
</div>
`;
